src/js/common/schemaform/reducers/index.js

```diff
@@ -17,16 +17,30 @@ import { SET_DATA,
 } from '../actions';
 
 function recalculateSchemaAndData(initialState) {
-  return Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
+  const retVal = Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = Object.keys(state).reduce((carry, pageName) => {
```

Comment:
`submission` never makes it to `formData` from my tests; it doesn't have `data`. Will there be `state.submission.data` at some point?